protocols/ledger/ledger.go

```diff
@@ -32,39 +32,44 @@ func (l *LedgerManager) HandleRequest(ledger *channel.TwoPartyLedger, request pr
 	if err != nil {
 		return protocols.SideEffects{}, fmt.Errorf("error finding a supported state: %w", err)
 	}
-
-	asset := types.Address{} // todo: loop over request.amount's assets
 	nextState := supported.Clone()
+	nextState.Outcome = outcome.Exit{}
+	// TODO: We should iterate over assets in RightAmount as well. For now we rely on the assumption that LeftAmount and RightAmount will have the same assets.
+	for asset := range request.LeftAmount {
+
+		currentLeftAmount := supported.Outcome.TotalAllocatedFor(request.Left)[asset]
+		currentRightAmount := supported.Outcome.TotalAllocatedFor(request.Right)[asset]
+		// Calculate the amounts
+		leftAmount := big.NewInt(0).Sub(currentLeftAmount, request.LeftAmount[asset])
+		rightAmount := big.NewInt(0).Sub(currentRightAmount, request.RightAmount[asset])
+		total := big.NewInt(0).Add(request.LeftAmount[asset], request.RightAmount[asset])
+		if leftAmount.Cmp(big.NewInt(0)) < 0 {
```

Comment:
You could reuse these utils here https://github.com/statechannels/go-nitro/blob/3f8b138550ba2ea7eb3966e2c5f7f98477549ac8/types/bigutils.go#L10

Reply:
Updated in [da0f79c](https://github.com/statechannels/go-nitro/pull/245/commits/da0f79c55e2eaa6a10cc72274441ea77c3cedbb3)